### PR TITLE
 Correct the HTML structure of the UI components to ensure validity

### PR DESCRIPTION
**Pull request by anagstef**
## Description

This PR alters some HTML elements to fix the validity of the HTML. This includes:
- Turning some `div` and `p` elements into `span`, especially the ones that are descendants of a `button`
- Include `aria-controls` and `aria-describedby` attributes only when there is a valid value
- Remove `height` and `width` attributes from an `img` element. Use CSS instead.
- Use correct roles and hierarchy for `tablist`, `menu`, `menuitem`.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ const _OrganizationSwitcher = withFloatingTree(() => {
           ref={reference}
           onClick={toggle}
           isOpen={isOpen}
-          aria-controls={switcherButtonMenuId}
+          aria-controls={isOpen ? switcherButtonMenuId : undefined}
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:29]: @anagstef are you sure ? maybe you simple wanna add `aria-expanded` ?

File: packages/clerk-js/src/ui/components/UserButton/UserButton.tsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ const _UserButton = withFloatingTree(() => {
         ref={reference}
         onClick={toggle}
         isOpen={isOpen}
-        aria-controls={userButtonMenuId}
+        aria-controls={isOpen ? userButtonMenuId : undefined}
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/UserButton/UserButton.tsx:29]: ditto

**Review by nikosdouvlis — APPROVED** [review]

File: packages/clerk-js/src/ui/elements/OrganizationPreview.tsx [hunk 3]
```diff
@@ -52,13 +52,15 @@ export const OrganizationPreview = (props: OrganizationPreviewProps) => {
       elementId={descriptors.organizationPreview.setId(elementId)}
       gap={3}
       align='center'
+      as='span'
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/elements/OrganizationPreview.tsx:55]: OrganizationPreview might not always be a child of a button, probably it's ok, i just wanted to clarify

File: packages/clerk-js/src/ui/primitives/Input.tsx [hunk 4]
```diff
@@ -95,7 +95,7 @@ export const Input = React.forwardRef<HTMLInputElement, InputProps>((props, ref)
       required={_required}
       id={props.id || fieldControlProps.id}
       aria-invalid={_hasError}
-      aria-describedby={errorMessageId}
+      aria-describedby={errorMessageId ? errorMessageId : undefined}
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/primitives/Input.tsx:98]: 💯

File: packages/clerk-js/src/ui/elements/OrganizationPreview.tsx (hunk 3 above)
Comment (anagstef) [reply packages/clerk-js/src/ui/elements/OrganizationPreview.tsx:55]: Yes, do you think there will be an issue with this being always a `span`?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx (hunk 1 above)
Comment (anagstef) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:29]: @panteliselef the `aria-controls` attribute needs to point to an element in HTML that exists in the DOM. The UserButton popover gets added to the DOM only when you click on the button.

File: packages/clerk-js/src/ui/elements/OrganizationPreview.tsx (hunk 3 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/elements/OrganizationPreview.tsx:55]: No issue here

**Review by panteliselef — APPROVED** [review]
lgtm

**Pull request merged**